hbase10/src/main/java/com/yahoo/ycsb/db/HBaseClient10.java

```diff
@@ -132,9 +132,11 @@ public void init() throws DBException {
     }
 
     try {
-      THREAD_COUNT.getAndIncrement();
       synchronized(THREAD_COUNT) {
-        connection = ConnectionFactory.createConnection(config);
+        THREAD_COUNT.getAndIncrement();
```

Comment:
I guess synchronize on the Connection object, since that will still be a proper Object shared wherever the THREAD_COUNT is.